dataprofiler/profilers/historical_profiler.py

```diff
@@ -0,0 +1,495 @@
+            "exclude_keys": [
+                "categories",
+                "gini_impurity",
+                "unalikeability",
+                "categorical_count",
+            ],
+        }
+        return default_opts
+
+    def report(self):
+        """Return the historical profiler report."""
+        return self.historical_profile
+
+    def _get_value_from_index_in_dict_list(self, d, index):
+        value_dict = {}
+        for key, val in d.items():
+            if isinstance(val, dict):
+                value_dict[key] = self._get_value_from_index_in_dict_list(val, index)
+            else:
+                value_dict[key] = copy.deepcopy(val[index])
+        return value_dict
+
+    def get_profile_report_by_index(self, index: int):
+        """Return the report of the profile stored at `index`."""
+        if index is None:
+            raise ValueError("`index` is `None`, expected `int` type")
+        if index < 0 or index >= self.length:
+            raise ValueError(
+                "`index`: {} out of bounds within this historical profiler".format(
+                    index
+                )
+            )
+
+        hp = self.historical_profile
+
+        profile_report = {}
+
+        profile_report["global_stats"] = self._get_value_from_index_in_dict_list(
+            hp["global_stats"], index
+        )
+
+        profile_report["data_stats"] = []
+        for col in hp["data_stats"]:
+            profile_report["data_stats"].append(
+                self._get_value_from_index_in_dict_list(col, index)
+            )
+
+        return profile_report
+
+    def get_most_recent_profile_report(self):
+        """Return the most recent profile report stored in this object."""
+        if self.historical_profile is None:
+            raise ValueError("This Historical Profiler has not been initialized")
+
+        return self.get_profile_report_by_index(0)
+
+    def get_oldest_profile_report(self):
+        """Return the oldest profile report stored in this object."""
+        if self.historical_profile is None:
+            raise ValueError("This Historical Profiler has not been initialized")
+
+        return self.get_profile_report_by_index((self.length - 1))
+
+    def _update_profile_values_in_dict_at_index(
+        self, list_dict: dict, values_dict: dict, index: int
+    ):
+        """
+        Insert individual values into list.
+
+        Takes the individual values stored in values_dict and
+        inserts them into the front of the list values stored in list_dict
+
+        This method will also remove the last value stored in the lists in list_dict
+        in the case that adding the new value will result in the list becoming
+        larger than `max_length` set in options.
+        """
+        for key, val in list_dict.items():
+            if key in self.options["exclude_keys"]:
+                continue
+            if key not in values_dict:
+                print(
```

Comment:
use logger like in profile_builder

Reply:
Realized this would be better accomplished as a warning